src/actions/reports/cost.py

```diff
@@ -84,23 +109,26 @@ def __str__(self):
 
 
 class PlanNodeCollector(PlanNodeVisitor):
-    def __init__(self, ctx, logger):
+    def __init__(self, ctx, plan_ctx, node_detail_map, logger):
         super().__init__()
         self.ctx = ctx
+        self.plan_ctx = plan_ctx
+        self.node_detail_map = node_detail_map
         self.logger = logger
         self.num_scans = 0
         self.depth = 0
+        self.scan_node_width_map = self.compute_scan_node_width(plan_ctx.get_query())
 
     def __enter(self):
         self.ctx.pf.__init__()
 
     def __exit(self):
         self.ctx.pf.has_single_scan_node = (self.num_scans == 1)
-        self.ctx.pf.has_no_condition_scan = (not self.ctx.pf.has_key_access_index
-                                             and not self.ctx.pf.has_scan_filter_index
-                                             and not self.ctx.pf.has_tfbr_filter_index
-                                             and not self.ctx.pf.has_table_filter_seqscan
-                                             and not self.ctx.pf.has_local_filter)
+        self.ctx.pf.has_no_condition_scan = not (self.ctx.pf.has_key_access_index
+                                                 or self.ctx.pf.has_scan_filter_index
+                                                 or self.ctx.pf.has_tfbr_filter_index
+                                                 or self.ctx.pf.has_table_filter_seqscan
+                                                 or self.ctx.pf.has_local_filter)
 
     def visit_PlanNode(self, node):
```

Comment:
Bad naming, same for visit_ScanNode. Should it be `visit_plan_node`?

Reply:
As you can see [here](https://github.com/yugabyte/taqo/blob/b4212069a91c510597d6c131bec0d210a2cb75ce/src/objects.py#L240), the base class calls "visit_" + class name of the node if it's defined or "generic_visit" otherwise, a technique used in [python ast](https://github.com/python/cpython/blob/79e479c45fc63b6001b206fec832064c31fc1f11/Lib/ast.py#L392).

Since I'm going to introduce JoinNode, etc. going forward, this approach is easier to maintain and less error-prone than typical hard coded visitors.

Reply:
Added the snake-casing regex